x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/lucene/ValueSources.java

```diff
@@ -71,19 +78,53 @@ public static List<ValueSourceInfo> sources(
             try {
                 fieldData = ctx.getForField(fieldType, MappedFieldType.FielddataOperation.SEARCH);
             } catch (IllegalArgumentException e) {
-                if (asUnsupportedSource) {
-                    sources.add(
+                switch (elementType) {
+                    case BYTES_REF -> sources.add(
                         new ValueSourceInfo(
                             new UnsupportedValueSourceType(fieldType.typeName()),
                             new UnsupportedValueSource(null),
                             elementType,
                             ctx.getIndexReader()
                         )
                     );
-                    continue;
-                } else {
-                    throw e;
+
+                    case LONG, INT -> sources.add(
+                        new ValueSourceInfo(CoreValuesSourceType.NUMERIC, ValuesSource.Numeric.EMPTY, elementType, ctx.getIndexReader())
+                    );
+
+                    case BOOLEAN -> sources.add(
+                        new ValueSourceInfo(CoreValuesSourceType.BOOLEAN, ValuesSource.Numeric.EMPTY, elementType, ctx.getIndexReader())
+                    );
+
+                    case DOUBLE -> sources.add(new ValueSourceInfo(CoreValuesSourceType.NUMERIC, new ValuesSource.Numeric() {
+                        @Override
+                        public boolean isFloatingPoint() {
+                            return true;
+                        }
+
+                        @Override
+                        public SortedNumericDocValues longValues(LeafReaderContext context) {
+                            return DocValues.emptySortedNumeric();
+                        }
+
+                        @Override
+                        public SortedNumericDoubleValues doubleValues(LeafReaderContext context) throws IOException {
+                            return org.elasticsearch.index.fielddata.FieldData.emptySortedNumericDoubles();
+                        }
+
+                        @Override
+                        public SortedBinaryDocValues bytesValues(LeafReaderContext context) throws IOException {
+                            return org.elasticsearch.index.fielddata.FieldData.emptySortedBinary();
+                        }
+
+                    }, elementType, ctx.getIndexReader()));
+                    default -> throw e;
                 }
+                HeaderWarning.addWarning(
+                    "Field [{}] cannot be retrieved, probably it is unsupported or not indexed; returning null",
```

Comment:
Why the `probably` word? Are there any other situations other than "unsupported" and "not indexed"?

Reply:
good question, these are the two situations we know as of now